### PR TITLE
Docs for lazy stream operators

### PR DESCRIPTION
**Pull request by johanandren**
References #25468

### CODE REVIEW

**Review by chbatey — APPROVED** [review]
Nice

File: akka-docs/src/main/paradox/stream/operators/Flow/lazyFlow.md [hunk 1]
```diff
@@ -11,11 +11,46 @@ Defers creation and materialization of a `Flow` until there is a first element.
 
 ## Description
 
-When the first element comes from upstream the actual `Flow` is created and materialized.
-The internal `Flow` will not be created if there are no elements on completion or failure of up or downstream.
+Defers `Flow` creation and materialization until when the first element arrives at the `lazyFlow` from upstream. After
+that the stream behaves as if the nested flow replaced the `lazyFlow`.
+The nested `Flow` will not be created if there are no elements on completion or failure of up or downstream.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Flow/lazyFlow.md:16]:
```suggestion
The nested `Flow` will not be created if the outer flow completes or fails before any elements arrive.
```

File: akka-docs/src/main/paradox/stream/operators/Flow/lazyFlow.md [hunk 2]
```diff
@@ -11,11 +11,46 @@ Defers creation and materialization of a `Flow` until there is a first element.
...
 
-The materialized value of the `Flow` will be the materialized value of the created internal flow if it is materialized
-and failed with a `akka.stream.NeverMaterializedException` if the stream fails or completes without the flow being materialized.
+Note that asynchronous boundaries and many other operators in the stream may do pre-fetching or trigger demand causing an element
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Flow/lazyFlow.md:18]: "causing an element"?

**Review by ennru — APPROVED** [review]
LGTM, just some minor language things.

**Pull request merged**